### PR TITLE
Update idioms.md with suggestion to use inline value classes

### PR DESCRIPTION
**Pull request by gw111zz**
Update idioms.md with suggestion to use inline value classes

### CODE REVIEW

**Comment by sarahhaggarty** [comment]
Hi @gw111zz 
Thanks for your suggestion! Could you move your idiom lower in the document so that it's underneath the "Create a singleton" idiom?

**Comment by gw111zz** [comment]
> Hi @gw111zz Thanks for your suggestion! Could you move your idiom lower in the document so that it's underneath the "Create a singleton" idiom?

Thanks for your comment @sarahhaggarty . I've made the suggested change.

File: docs/topics/idioms.md [hunk 1]
```diff
@@ -125,6 +125,21 @@ object Resource {
 }
 ```
 
+## Use inline value classes to wrap single values for type-safety
```
Comment (sarahhaggarty) [inline docs/topics/idioms.md:128]:
```suggestion
## Use inline classes for type-safe values
```

File: docs/topics/idioms.md [hunk 2]
```diff
@@ -125,6 +125,21 @@ object Resource {
...
+```kotlin
+@JvmInline
+value class EmployeeId(private val id: String)
+
+@JvmInline
+value class CustomerId(private val id: String)
+```
+
+Now accidentally mixing up `EmployeeId` and `CustomerId` will result in a compilation error. The inline classes will be represented by the 
+underlying type at runtime. `@JvmInline` is only needed for JVM backends.
```
Comment (sarahhaggarty) [inline docs/topics/idioms.md:139]:
```suggestion
If you accidentally mix up `EmployeeId` and `CustomerId`, a compilation error is triggered.

> The `@JvmInline` annotation is only needed for JVM backends.
>
{type="note"}
```

File: docs/topics/idioms.md [hunk 3]
```diff
@@ -125,6 +125,21 @@ object Resource {
...
+
+(See [Inline classes](inline-classes.md) for more information)
```
Comment (sarahhaggarty) [inline docs/topics/idioms.md:141]:
```suggestion
For more information, see [Inline value classes](inline-classes.md).
```

**Review by sarahhaggarty — CHANGES_REQUESTED** [review]
Thanks for your suggestion @gw111zz !
I've suggested some changes to bring the text in line with our style guide. Please can you apply them? Then we can go ahead with the merge! 🚀

**Comment by sarahhaggarty** [comment]
Hi @gw111zz 
Are you able to implement these changes? If I don't hear from you within a week, I'll combine these changes into another PR.

**Comment by sarahhaggarty** [comment]
Changes add in #4088!

**Pull request closed without merging**